### PR TITLE
noloc for <xref href=...> and <xref uid=...>

### PR DESCRIPTION
**Pull request by AngryBerryMS**
[AB#532860](https://ceapex.visualstudio.com/Engineering/_workitems/edit/532860)

### CODE REVIEW

File: src/docfx/lib/HtmlUtility.cs [hunk 1]
```diff
@@ -512,4 +514,7 @@ private static string AddLocaleIfMissingForAbsolutePath(string href, string loca
     private static bool IsSelfClosingElement(string tagName) => s_selfClosingTags.Contains(tagName.ToLowerInvariant());
 
     private static bool IsSelfClosingElement(this HtmlToken token) => IsSelfClosingElement(token.Name.ToString());
+
+    private static string AddNoLocSpan(string? content)
+        => $"<span class=\"no-loc\">{content}</span>";
```
Comment (herohua) [inline src/docfx/lib/HtmlUtility.cs:519]: put inside `TransformXref` block?

**Review by herohua — APPROVED** [review]
:shipit:

File: docs/specs/toc/toc.yml [hunk 2]
```diff
@@ -1435,3 +1435,16 @@ outputs:
   a.mta.json:
   toc.json: |
     {"items":[{"name":"a","href":"a"}],"_path":"toc.json","schema":"toc","outputType":"pagejson"}
+---
+# unexist xref in toc
```
Comment (live1206) [inline docs/specs/toc/toc.yml:1439]: consider to put this test in `xref.yml`

Comment (live1206) [inline docs/specs/toc/toc.yml:1439]:
```suggestion
# unkowned xref in toc
```

File: src/docfx/lib/HtmlUtility.cs (hunk 1 above)
Comment (AngryBerryMS) [reply src/docfx/lib/HtmlUtility.cs]: Done

File: docs/specs/toc/toc.yml (hunk 2 above)
Comment (AngryBerryMS) [reply docs/specs/toc/toc.yml]: Done

Comment (AngryBerryMS) [reply docs/specs/toc/toc.yml]: Fixed.

**Review by live1206 — APPROVED** [review]

File: src/docfx/lib/HtmlUtility.cs [hunk 3]
```diff
@@ -278,17 +278,22 @@ public static void TransformLink(ref HtmlToken token, MarkdownObject? block, Fun
...
         string GetDefaultResolvedNode()
         {
             var content = !string.IsNullOrEmpty(xrefLink.Display) ?
                 xrefLink.Display : (href != null ? UrlUtility.SplitUrl(href).path : uid);
-            return StringUtility.Html($"<span class=\"xref\">{content}</span>");
+            return StringUtility.Html($"<span class=\"{(xrefLink.Localizable ? string.Empty : "no-loc ")}xref\">{content}</span>");
         }
+
+        string AddNoLocSpan(string? content)
+            => $"<span class=\"no-loc\">{content}</span>";
```
Comment (yufeih) [inline src/docfx/lib/HtmlUtility.cs:296]: Add `static`.
```suggestion
        static string AddNoLocSpan(string? content) => $"<span class=\"no-loc\">{content}</span>";
```

**Review by yufeih — APPROVED** [review]

**Pull request merged**